projects/lookup-showcase/src/app/visual/lookup/lookup-visual.component.ts

```diff
@@ -132,6 +136,30 @@ export class LookupVisualComponent implements OnInit {
       this.bestFriendSelectMode === 'single' ? 'multiple' : 'single';
   }
 
+  public bestFriendSearch(args: SkyAutocompleteSearchAsyncArgs) {
+    const searchText = (args.searchText || '').toLowerCase();
+
+    let items = this.people.filter(
+      (item) => item.name.toLowerCase().indexOf(searchText) >= 0
+    );
+
+    const totalCount = items.length;
+    let hasMore = false;
+    let itemCountToReturn = args.displayType === 'popover' ? 5 : 10;
```

Comment:
Since data loads asynchronously and all "load more" logic results in a call back to the consumer, it's up to the consumer to determine how many records to display. Otherwise we'd end up with double logic like this:

- Consumer returns 15 records
- Lookup decides to only show 10
- User scrolls down to load more records
- Lookup already has the next 5 records, so it asks consumer for the next records
- Consumer returns another 15 records
- User scrolls down to load more records
- Lookup has 10 records left over from the next call, so it doesn't call the consumer again

All that seems too complicated and results in more records being returned than are needed, so it's best to leave the number of displayed records up to the consumer based on how the records are being displayed.